aten/src/ATen/GeneratorForPrivareuseone.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <c10/util/intrusive_ptr.h>
+
+namespace at {
```

Comment:
We don't want to bother adding unnecessary symbols to the public `at` namespace. Can you rename the `Register` class to something like `_GeneratorRegisterer`, and then move the static hashmap and its related code into a cpp file (you can also put it in an anonymous namespace)?